mlflow/tracking/client.py

```diff
@@ -1631,7 +1631,7 @@ def log_dict(self, run_id: str, dictionary: Dict[str, Any], artifact_file: str)
                 if extension in [".yml", ".yaml"]:
                     yaml.dump(dictionary, f, indent=2, default_flow_style=False)
                 else:
-                    json.dump(dictionary, f, indent=2)
+                    json.dump(dictionary, f, indent=2, default=str)
```

Comment:
Can we update the docstring and mention we stringify objects that can't be JSON-serialized?